index.js

```diff
@@ -2,84 +2,69 @@ const path = require('path');
-  },
-  'yarn/yarn-offline': {
-    average: 0,
-    name: 'yarn --offline',
-    runs: []
-  },
-  'npm5/npm': {
-    average: 0,
-    name: 'npm 5.x',
-    runs: []
-  },
-  'npm5/npm-cached': {
-    average: 0,
-    name: 'npm 5.x (cached)',
-    runs: []
-  },
-  'npm5/shrinkpack': {
-    average: 0,
-    name: 'npm 5.x (shrinkpacked)',
-    runs: []
-  },
-  'npm5/shrinkpack-compressed': {
-    average: 0,
-    name: 'npm 5.x (shrinkpacked, compressed)',
-    runs: []
-  }
+  'npm/npm': { average: 0, name: 'npm 4.x', runs: [] },
+  'npm/npm-cached': { average: 0, name: 'npm 4.x (cached)', runs: [] },
+  'npm/shrinkpack': { average: 0, name: 'npm 4.x (shrinkpacked)', runs: [] },
+  'npm/shrinkpack-compressed': { average: 0, name: 'npm 4.x (shrinkpacked, compressed)', runs: [] },
+  'yarn/yarn': { average: 0, name: 'yarn', runs: [] },
+  'yarn/yarn-offline': { average: 0, name: 'yarn --offline', runs: [] },
+  'npm5/npm': { average: 0, name: 'npm 5.x', runs: [] },
+  'npm5/npm-cached': { average: 0, name: 'npm 5.x (cached)', runs: [] },
+  'npm5/shrinkpack': { average: 0, name: 'npm 5.x (shrinkpacked)', runs: [] },
+  'npm5/shrinkpack-compressed': { average: 0, name: 'npm 5.x (shrinkpacked, compressed)', runs: [] },
+  'pnpm/pnpm': { average: 0, name: 'pnpm', runs: [] },
+  'pnpm/pnpm-cached': { average: 0, name: 'pnpm (cached)', runs: [] },
+  'pnpm/pnpm-offline': { average: 0, name: 'pnpm --offline', runs: [] },
 };
 
-function average(list) {
-  return list.reduce((total, value) => total + value, 0) / list.length;
+const cacheClean = {
+  pnpm: function(dirPath) {
+    spawn('rm', ['-rf', path.join(dirPath, '.pnpm-registry')]);
+    spawn('rm', ['-rf', path.join(dirPath, '.pnpm-store')]);
+  }
 }
 
-function spawn(cmd, args, options = {}) {
-  childProcess.spawnSync(cmd, args, options);
+const cacheCleanArgs = ['cache', 'clean'];
+
+try {
+  runAll();
+} catch (err) {
+  onError(err);
 }
 
-function clean(dirPath) {
-  spawn('npm', ['cache', 'clean']);
-  spawn('yarn', ['cache', 'clean']);
-  if (dirPath.includes('-cached') === false) {
-    spawn('rm', ['-rf', path.join(dirPath, 'node_cache')]);
+// implementation
+
+function runAll() {
+  const sampleSize = 5;
+  for (let i = 1; i <= sampleSize; i++) {
+    console.log(`Run ${i}`);
+    runBenchmark('npm', ['install'], 'npm');
+    runBenchmark('npm', ['install'], 'npm-cached');
+    runBenchmark('npm', ['install'], 'shrinkpack');
+    runBenchmark('npm', ['install'], 'shrinkpack-compressed');
+    runBenchmark('yarn', ['install'], 'yarn');
+    runBenchmark('yarn', ['install', '--offline'], 'yarn-offline');
+    runBenchmark('npm5', ['install'], 'npm');
+    runBenchmark('npm5', ['install'], 'npm-cached');
+    runBenchmark('npm5', ['install'], 'shrinkpack');
+    runBenchmark('npm5', ['install'], 'shrinkpack-compressed');
+    runBenchmark('pnpm', ['install'], 'pnpm');
+    runBenchmark('pnpm', ['install'], 'pnpm-cached');
+    runBenchmark('pnpm', ['install'], 'pnpm-offline');
```

Comment:
I originally had it and removed it, because I thought the npmrc file was enough to force it